Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java

```diff
@@ -779,19 +786,31 @@ private static DateTime setEndValidTime(DateTime endValidTimeArg, DateTime issue
         if (StringUtils.isNotEmpty(acp)) {
             acpAttribute = OpenSAML2ComponentBuilder.getInstance()
                     .createAttribute(SamlConstants.ATTRIBUTE_FRIENDLY_NAME_XUA_ACP, SamlConstants.ATTRIBUTE_NAME_XUA_ACP,
-                            SamlConstants.URI_NAME_FORMAT, Arrays.asList(acp));
+                            SamlConstants.URI_NAME_FORMAT);
+            acpAttribute.getAttributeValues().add(createURI(acp));
             statements
                     .addAll(OpenSAML2ComponentBuilder.getInstance().createAttributeStatement(Arrays.asList(acpAttribute)));
         }
         if(StringUtils.isNotEmpty(iacp)) {
             iacpAttribute = OpenSAML2ComponentBuilder.getInstance()
                     .createAttribute(SamlConstants.ATTRIBUTE_FRIENDLY_NAME_XUA_IACP, SamlConstants.ATTRIBUTE_NAME_XUA_IACP,
-                            SamlConstants.URI_NAME_FORMAT, Arrays.asList(iacp));
+                            SamlConstants.URI_NAME_FORMAT);
+            iacpAttribute.getAttributeValues().add(createURI(iacp));
             statements
                     .addAll(OpenSAML2ComponentBuilder.getInstance().createAttributeStatement(Arrays.asList(iacpAttribute)));
         }
         return statements;
     }
+    
+    private XMLObject createURI(String value) {
```

Comment:
Why do we create the method in this class instead of OpenSAML2ComponentBuilder?  OpenSAML2ComponentBuilder is probably the right class to handle all creation saml attribute statement and value

Reply:
Updating.  Will submit another PR.